### PR TITLE
E2E Tests: Bump `efficientgo/e2e` version

### PR DESCRIPTION
**Pull request by matej-g**
Signed-off-by: Matej Gera <matejgera@gmail.com>

* [ ] I added CHANGELOG entry for this change.
* [X] Change is not relevant to the end user.

## Changes

Bumping the `efficientgo/e2e` to the latest `main` - this includes some refactoring changes (no effect on test functionality) and using the new HTTPS readiness probe for `minio`.

## Verification

Running E2E tests

### CODE REVIEW

**Review by saswatamcode — APPROVED** [review]
Thanks! This looks awesome! 💫

**Review by GiedriusS — APPROVED** [review]

**Comment by GiedriusS** [comment]
Good work :1st_place_medal: !

File: test/e2e/receive_test.go [hunk 1]
```diff
@@ -496,9 +496,9 @@ func TestReceive(t *testing.T) {
 
 		h := receive.HashringConfig{
 			Endpoints: []string{
-				r1.InternalEndpoint("grpc"),
-				r2.InternalEndpoint("grpc"),
-				r3.InternalEndpoint("grpc"),
+				r1.Future().InternalEndpoint("grpc"),
```
Comment (bwplotka) [inline test/e2e/receive_test.go:499]: I did not design it to do Future() all the time, so maybe r1 = r1.Future would make more sense, but not a big deal

**Comment by bwplotka** [comment]
Thanks a lot!

**Pull request merged**